slatedb/src/compactions_store.rs

```diff
@@ -0,0 +1,495 @@
+use crate::clock::SystemClock;
+use crate::compactor_state::Compactions;
+use crate::error::SlateDBError;
+#[allow(dead_code)]
+use crate::error::SlateDBError::LatestTransactionalObjectVersionMissing;
+use crate::flatbuffer_types::FlatBufferCompactionsCodec;
+use crate::transactional_object::object_store::ObjectStoreSequencedStorageProtocol;
+use crate::transactional_object::{
+    DirtyObject, FenceableTransactionalObject, MonotonicId, SequencedStorageProtocol,
+    SimpleTransactionalObject, TransactionalObject, TransactionalStorageProtocol,
+};
+use chrono::Utc;
+use object_store::path::Path;
+use object_store::ObjectStore;
+use serde::Serialize;
+#[cfg(test)]
+use std::ops::RangeBounds;
+use std::sync::Arc;
+use std::time::Duration;
+
+/// Represents the compactions stored in the object store. This type tracks the current
+/// contents and id of the stored compactions state, and allows callers to read and update
+/// it using transactional semantics.
+pub(crate) struct StoredCompactions {
```

Comment:
just for my understanding, is the reason we make this `pub(crate)` instead of just requiring a `FenceableCompactions` because admin tooling will use this eventually (the main compactor should always use the fencing version)?

Reply:
That is a great question. I just cargo culted from `manifest/store.rs` :P But yea, after checking "find all references", I can see we use the manifest one admin.rs and quite a few other spots as well. Right now, `StoredCompactions` is only used in compactor.rs, so I suppose we could go with a pub(mod) approach, but we don't have a mod for `compactor` right now (we should). I'm inclined to leave as-is since GC will probably need this and so will admin.rs stuff when we add compaction trigger commands.